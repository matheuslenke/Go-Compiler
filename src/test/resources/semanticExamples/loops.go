package samples

import "fmt"


func ForStmts() {
	var a,b int
	var c [1]int
	a, b = 1, 2
	for a < b {
		fmt.Println("From condition-only ForStmt")
		a = a + 1

		for (b < 3) {
			fmt.Println("From condition-only ForStmt")
			b = b + 1
		}
	}

	for a <= b {
		fmt.Println("For running")
		a = a + 1
	}

	for a == a {
		fmt.Println("Infinity loop")
	}
}
