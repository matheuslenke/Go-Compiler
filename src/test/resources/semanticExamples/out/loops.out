[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:
Entry 0: From condition-only ForStmt
Entry 1: For running
Entry 2: Infinity loop




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: ForStmts, line: 6, type: func, scopeName: global
Entry 6 -- name: a, line: 7, type: int, scopeName: ForStmts
Entry 7 -- name: b, line: 7, type: int, scopeName: ForStmts
Entry 8 -- name: c, line: 8, type: array, subtype: int, scopeName: ForStmts




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.975 s
[INFO] Finished at: 2022-02-24T20:53:57-03:00
[INFO] ------------------------------------------------------------------------
